src/foundry/client-esm/canvas/sources/base-light-source.d.mts

```diff
@@ -69,12 +73,18 @@ declare class BaseLightSource<
 
   override _updateCommonUniforms(shader: AbstractBaseShader): void;
 
+  /** @remarks This property is undocumented, and only defined during `_updateCommonUniforms` */
+  cachededAttentuation?: number;
```

Comment:
Is it possible for any of these properties to be explicitly undefined?

Reply:
Nope. They only have two references each, anywhere in foundry, and they're all here in `BaseLightSource#_updateCommonUniforms`:
```js
if ( this.cachedAttenuation !== this.data.attenuation ) {
  this.computedAttenuation = (Math.cos(Math.PI * Math.pow(this.data.attenuation, 1.5)) - 1) / -2;
  this.cachedAttenuation = this.data.attenuation;
}
u.attenuation = this.computedAttenuation;
```